### ISSUE TITLE
Implement early stopping

### ISSUE THREAD

**Issue by rabyj** [laperlej/epi_ml#3]
Currently the number of epoch is fixed. We want to implement a convergence criteria that permits the training to be stopped before.

**Comment by rabyj** [comment]
Early stopping implemented with the use of checkpoints/saver. The criteria used is the number of epochs since last maximum of validation accuracy.
